// kickstart-node/js/kickstart.js
(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({"./lib-core/coffee/app.coffee":[function(require,module,exports){
var Buffer, Buttons, Debounce, Dropdown, Growl, Icons, KS, Modal, Navbar, Status, Tabs, Throttler;

KS = require('./ks');

Modal = require('./modal');

Navbar = require('./navbar');

Debounce = require('./debouncer');

Icons = require('./icons');

Status = require('./status');

Tabs = require('./tabs');

Throttler = require('./throttler');

Buttons = require('./buttons');

Buffer = require('./buffer');

Growl = require('./growl');

Dropdown = require('./dropdown');

k$.ready = function() {
  var $navbar, $tabSet, _i, _j, _len, _len1, _ref, _ref1, _results;
  k$.icons();
  k$.button();
  k$.dropdown();
  _ref = k$.$$('[data-ks-navbar]');
  for (_i = 0, _len = _ref.length; _i < _len; _i++) {
    $navbar = _ref[_i];
    k$.nav($navbar);
  }
  _ref1 = k$.$$('[data-ks-tabs]');
  _results = [];
  for (_j = 0, _len1 = _ref1.length; _j < _len1; _j++) {
    $tabSet = _ref1[_j];
    _results.push(k$.tabs($tabSet));
  }
  return _results;
};

document.addEventListener('DOMContentLoaded', function() {
  return k$.ready();
});



},{"./buffer":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/buffer.coffee","./buttons":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/buttons.coffee","./debouncer":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/debouncer.coffee","./dropdown":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/dropdown.coffee","./growl":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/growl.coffee","./icons":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/icons.coffee","./ks":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/ks.coffee","./modal":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/modal.coffee","./navbar":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/navbar.coffee","./status":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/status.coffee","./tabs":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/tabs.coffee","./throttler":"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/throttler.coffee"}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/buffer.coffee":[function(require,module,exports){
var buffer;

buffer = function(fn, delay) {
  var i;
  k$.bufferArray = k$.bufferArray || new Array();
  if (!k$.bufferArray.length) {
    k$.bufferArray = new Array();
    delay = delay || 500;
    i = 1;
    k$.bufferInterval = setInterval(function() {
      if (k$.bufferArray[i]) {
        k$.bufferArray[i]();
      }
      i++;
      if (i >= k$.bufferArray.length) {
        clearInterval(k$.bufferInterval);
        k$.bufferArray = void 0;
        return i = 1;
      }
    }, delay);
  }
  k$.bufferArray.push(fn);
  if (k$.bufferArray.length === 1) {
    k$.bufferArray[0]();
  }
  return console.info("Function queued (" + k$.bufferArray.length + " in queue)");
};

k$.buffer = buffer;

module.exports = buffer;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/buttons.coffee":[function(require,module,exports){
var button;

button = function() {
  var $button, $buttonDropdown, _i, _j, _len, _len1, _ref, _ref1, _results;
  _ref = k$.$$("button");
  for (_i = 0, _len = _ref.length; _i < _len; _i++) {
    $button = _ref[_i];
    if ($button.querySelectorAll('ul').length) {
      $button.classList.add('menu-item');
    }
  }
  _ref1 = k$.$$('.button-dropdown');
  _results = [];
  for (_j = 0, _len1 = _ref1.length; _j < _len1; _j++) {
    $buttonDropdown = _ref1[_j];
    _results.push($buttonDropdown.parentNode.classList.add('menu-item'));
  }
  return _results;
};

k$.button = button;

module.exports = button;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/debouncer.coffee":[function(require,module,exports){
var debounce;

debounce = function(fn, id, delay) {
  var $delay;
  $delay = delay || 1000;
  if (k$.debounceQueue === null) {
    k$.debounceQueue = id;
  }
  if (id === k$.debounceQueue) {
    clearTimeout(k$.debounceTimer);
  }
  return k$.debounceTimer = setTimeout(function() {
    fn();
    return k$.debounceQueue = null;
  }, $delay);
};

k$.debounce = debounce;

module.exports = debounce;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/dropdown.coffee":[function(require,module,exports){
var dropdown;

dropdown = function() {
  return document.body.addEventListener('click', function(e) {
    var $menuItem, closeAllMenus, openMenu;
    $menuItem = null;
    closeAllMenus = function() {
      var _$menuItem, _i, _len, _ref, _results;
      _ref = k$.$$('.menu-item');
      _results = [];
      for (_i = 0, _len = _ref.length; _i < _len; _i++) {
        _$menuItem = _ref[_i];
        _results.push(_$menuItem.classList.remove('open'));
      }
      return _results;
    };
    openMenu = function() {
      if ($menuItem.classList.contains('open')) {
        $menuItem.classList.remove('open');
      } else {
        closeAllMenus();
        $menuItem.classList.add('open');
      }
      return e.stopPropagation();
    };
    if (e.target.classList.contains('menu-item')) {
      $menuItem = e.target;
      return openMenu();
    } else if (e.target.parentNode.classList.contains('menu-item')) {
      $menuItem = e.target.parentNode;
      return openMenu();
    } else {
      closeAllMenus();
    }
  });
};

k$.dropdown = dropdown;

module.exports = dropdown;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/growl.coffee":[function(require,module,exports){
var growl;

growl = function(params) {
  return k$.buffer(function() {
    var className, content, defaults, delay, growlContainer, id;
    defaults = {
      title: void 0,
      text: void 0,
      delay: 2000,
      type: 'growl-warn',
      id: Date.now()
    };
    params = k$.extend(defaults, params);
    if (!k$.$$('.growl_container').length) {
      growlContainer = document.createElement('div');
      growlContainer.className = 'growl_container';
      document.body.appendChild(growlContainer);
    }
    growl = document.createElement('div');
    className = "alert growl show " + params.type + " growl-" + params.id;
    growl.className = className;
    content = "";
    if (params.title) {
      content += "<h1>" + params.title + "</h1>";
    }
    if (params.text) {
      content += "<p>" + params.text + "</p>";
    }
    growl.innerHTML = content;
    k$.$('.growl_container').appendChild(growl);
    delay = params.delay;
    id = params.id;
    if (delay > 0) {
      return (function(delay, id) {
        return setTimeout(function() {
          var $growl, $newGrowl;
          $growl = k$.$(".growl-" + id);
          $growl.classList.remove('show');
          $newGrowl = $growl.cloneNode(true);
          $growl.parentNode.replaceChild($newGrowl, $growl);
          $newGrowl.classList.add('hide');
          return (function(delay, id) {
            return setTimeout(function() {
              if (!k$.$$('.growl.show').length) {
                return k$.$('.growl_container').parentNode.removeChild(k$.$('.growl_container'));
              }
            }, 500);
          })(delay, id);
        }, delay);
      })(delay, id);
    }
  });
};

k$.growl = growl;

module.exports = growl;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/icons.coffee":[function(require,module,exports){
var icons;

icons = function() {
  var CACHE, IE9TO11, embed, onframe, onload, requestAnimationFrame, uses;
  uses = document.getElementsByTagName("use");
  requestAnimationFrame = window.requestAnimationFrame || window.setTimeout;
  CACHE = {};
  IE9TO11 = true;
  embed = function(svg, g) {
    var clone, fragment, viewBox;
    if (g) {
      viewBox = g.getAttribute("viewBox");
      fragment = document.createDocumentFragment();
      clone = g.cloneNode(true);
      if (viewBox) {
        svg.setAttribute("viewBox", viewBox);
      }
      while (clone.childNodes.length) {
        fragment.appendChild(clone.childNodes[0]);
      }
      svg.appendChild(fragment);
    }
  };
  onload = function() {
    var s, x, xhr;
    xhr = this;
    x = document.createElement("x");
    s = xhr.s;
    x.innerHTML = xhr.responseText;
    xhr.onload = function() {
      s.splice(0).map(function(array) {
        embed(array[0], x.querySelector("#" + array[1].replace(/(\W)/g, "\\$1")));
      });
    };
    xhr.onload();
  };
  onframe = function() {
    var svg, url, url_hash, url_root, use, xhr;
    use = void 0;
    while ((use = uses[0])) {
      svg = use.parentNode;
      url = use.getAttribute("xlink:href").split("#");
      url_root = url[0];
      url_hash = url[1];
      svg.removeChild(use);
      if (url_root.length) {
        xhr = CACHE[url_root] = CACHE[url_root] || new XMLHttpRequest();
        if (!xhr.s) {
          xhr.s = [];
          xhr.open("GET", url_root);
          xhr.onload = onload;
          xhr.send();
        }
        xhr.s.push([svg, url_hash]);
        if (xhr.readyState === 4) {
          xhr.onload();
        }
      } else {
        embed(svg, document.getElementById(url_hash));
      }
    }
    requestAnimationFrame(onframe);
  };
  if (IE9TO11) {
    onframe();
  }
};

k$.icons = icons;

module.exports = icons;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/ks.coffee":[function(require,module,exports){
(function (global){
global.k$ = new Object();

k$.$$ = function(el) {
  return document.querySelectorAll(el);
};

k$.$ = function(el) {
  return k$.$$(el)[0];
};

k$.debounceTimer = false;

k$.debounceQueue = null;

k$.extend = function(destination, source) {
  var property;
  for (property in source) {
    if (source[property] && source[property].constructor && source[property].constructor === Object) {
      destination[property] = destination[property] || {};
      arguments.callee(destination[property], source[property]);
    } else {
      destination[property] = source[property];
    }
  }
  return destination;
};

module.exports = k$;



}).call(this,typeof global !== "undefined" ? global : typeof self !== "undefined" ? self : typeof window !== "undefined" ? window : {})

},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/modal.coffee":[function(require,module,exports){
var modal;

modal = function(el) {
  var iOS = /(iPad|iPhone|iPod)/g.test( navigator.userAgent );
  if (iOS) {
    document.body.classList.add('dismiss-modal');
  }
  (function(el) {
    var $closer, $hideModal;
    $hideModal = function() {
      return k$.$(el).style.display = 'none';
    };
    document.body.addEventListener('click', function() {
      return $hideModal();
    });
    k$.$(el).addEventListener('click', function(e) {
      return e.stopPropagation();
    });
    $closer = k$.$(el).querySelector('a[data-modal-close]');
    if ($closer) {
      return $closer.addEventListener('click', function() {
        return $hideModal();
      });
    }
  })(el);
  return k$.$(el);
};

k$.modal = modal;

module.exports = modal;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/navbar.coffee":[function(require,module,exports){
var nav;

nav = function(el) {
  var $button, $menuItem, $menuItems, $navbar, e, _$menuItems, _i, _j, _len, _len1;
  $navbar = typeof el === 'string' ? k$.$(el) : el;
  try {
    $menuItems = $navbar.querySelectorAll('ul > li');
    _$menuItems = new Array();
    for (_i = 0, _len = $menuItems.length; _i < _len; _i++) {
      $menuItem = $menuItems[_i];
      if ($menuItem.querySelectorAll('ul').length && !$menuItem.querySelectorAll('[role="button"]').length) {
        _$menuItems.push($menuItem);
      }
    }
    $menuItems = _$menuItems;
    for (_j = 0, _len1 = $menuItems.length; _j < _len1; _j++) {
      $menuItem = $menuItems[_j];
      $menuItem.classList.add('menu-item');
    }
  } catch (_error) {
    e = _error;
    console.error("Could not instantiate as a nav.", e.message);
  }
  $button = $navbar.querySelector('.navbar-title button');
  if ($button) {
    return $button.addEventListener('click', function() {
      var $nav;
      $nav = $navbar.querySelector('nav');
      if ($nav.classList.contains('expand')) {
        return $nav.classList.remove('expand');
      } else {
        return $nav.classList.add('expand');
      }
    });
  }
};

k$.nav = nav;

module.exports = nav;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/status.coffee":[function(require,module,exports){
var status;

status = function(opts) {
  var $status, $statusBar, defaults, hideStatusBar;
  defaults = {
    type: 'status-yellow',
    delay: 2000
  };
  status = k$.extend(defaults, opts);
  if (!k$.$$('#status_bar').length) {
    $statusBar = document.createElement('div');
    $statusBar.id = 'status_bar';
    $statusBar.className = 'status_bar';
    $statusBar.innerHTML = "<div class='status_bar-status' id='status_bar-status'></div>";
    document.body.appendChild($statusBar);
  }
  $statusBar = k$.$('#status_bar');
  hideStatusBar = function() {
    $statusBar.classList.add('hide');
    return setTimeout(function() {
      $statusBar.classList.remove('hide');
      return $statusBar.parentNode.removeChild($statusBar);
    }, 250);
  };
  if (status.delay > 0) {
    k$.debounce(hideStatusBar, 'hideStatusBar', status.delay);
  }
  $status = k$.$("#status_bar-status");
  $status.innerHTML = status.text;
  return $status.dataset.type = status.type;
};

k$.status = status;

module.exports = status;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/tabs.coffee":[function(require,module,exports){
var tabs;

tabs = function(el) {
  var $_tab, $id, $pane, $paneSet, $tab, $tabContainer, $tabLink, $tabSet, _i, _j, _k, _len, _len1, _len2, _results;
  $tabContainer = typeof el === 'string' ? k$.$(el) : el;
  $tabSet = $tabContainer.querySelectorAll('li');
  for (_i = 0, _len = $tabSet.length; _i < _len; _i++) {
    $tab = $tabSet[_i];
    $tab.classList.add('tab-item');
  }
  $paneSet = new Array();
  for (_j = 0, _len1 = $tabSet.length; _j < _len1; _j++) {
    $_tab = $tabSet[_j];
    $id = $_tab.querySelector('a').getAttribute('href');
    $pane = k$.$("article" + $id);
    if ($_tab.classList.contains('open')) {
      $pane.classList.add('open');
    }
    $paneSet.push($pane);
    $pane.setAttribute('data-panel', true);
  }
  _results = [];
  for (_k = 0, _len2 = $tabSet.length; _k < _len2; _k++) {
    $tab = $tabSet[_k];
    $tabLink = $tab.querySelector('a');
    $tabLink.setAttribute('data-link', $tabLink.getAttribute('href'));
    $tabLink.href = 'javascript:void(0);';
    _results.push((function($tab, $tabLink, $paneSet) {
      return $tab.addEventListener('click', function() {
        var _$tab, _l, _len3, _len4, _m;
        for (_l = 0, _len3 = $paneSet.length; _l < _len3; _l++) {
          $pane = $paneSet[_l];
          $pane.classList.remove('open');
        }
        for (_m = 0, _len4 = $tabSet.length; _m < _len4; _m++) {
          _$tab = $tabSet[_m];
          _$tab.classList.remove('open');
        }
        k$.$("article" + ($tabLink.getAttribute('data-link'))).classList.add('open');
        return $tab.classList.add('open');
      });
    })($tab, $tabLink, $paneSet));
  }
  return _results;
};

k$.tabs = tabs;

module.exports = tabs;



},{}],"/Users/adamkochanowicz/Sites/kickstart/lib-core/coffee/throttler.coffee":[function(require,module,exports){
var throttle;

throttle = function(fn, id, delay) {};

k$.throttle = throttle;

module.exports = throttle;



},{}]},{},["./lib-core/coffee/app.coffee"])
